docs/reference/mapping/types/dense-vector.asciidoc

```diff
@@ -45,5 +45,130 @@ PUT my-index-000001/_doc/2
 }
 
 --------------------------------------------------
+<1> `dims` is a required parameter that defines the number of vector dimensions
 
-<1> dims – the number of dimensions in the vector, required parameter.
+NOTE:  Unlike most other data types, dense vectors are always single-valued.
+It is not possible to store multiple values in one `dense_vector` field.
+
+==== Indexing vectors for kNN search
+
+experimental[]
+
+_k-nearest neighbor search_ or _kNN_ is the task of finding the k nearest
+vectors to a query vector, as measured by a similarity metric.
+
+As mentioned above, `dense_vector` fields can be used to rank documents
+through <<query-dsl-script-score-query,`script_score` queries>>. This lets you
+perform a brute-force kNN search by scanning all documents and ranking them by
+similarity.
+
+In many cases, a brute-force kNN search is not efficient enough. For this
+reason, the `dense_vector` type supports indexing vectors into a specialized
+data structure to support fast kNN search. You can enable indexing by setting
+through the `index` parameter:
+
+[source,console]
+--------------------------------------------------
+PUT my-index-000002
+{
+  "mappings": {
+    "properties": {
+      "my_vector": {
+        "type": "dense_vector",
+        "dims": 3,
+        "index": true,
+        "similarity": "dot_product" <1>
+      }
+    }
+  }
+}
+--------------------------------------------------
+<1> When `index` is enabled, you must define the vector similarity to use in kNN search
+
+Elasticsearch uses the https://arxiv.org/abs/1603.09320[HNSW algorithm] to
+support efficient kNN search. Like most kNN algorithms, HNSW is an approximate
+method that sacrifices result accuracy for improved speed.
+
+NOTE: indexing vectors for kNN search is an expensive process. It can take
+substantial time to ingest documents that contain vector fields with `index`
+enabled.
+
+[discrete]
+[[dense-vector-params]]
+==== Parameters for dense vector fields
```

Comment:
I struggled with the formatting a bit here -- any corrections/ suggestions are appreciated. The other field types use the `[horizontal]` styling, but I couldn't get this to work with the sublists and code snippets.

Reply:
Thanks! I went through and accepted all of these (batched into a new commit).